### PR TITLE
Create Player Class

### PR DESCRIPTION
**Pull request by glitchwolfe**

### CODE REVIEW

File: Assets/Scripts/Player/Player.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	public Deck deck;
+	public List<Card> hand;
+
+	public int startingHandSize = 4;
+	public int maxHandSize;
+
+	// Events!
+	// @TODO: Figure out how to implement these as GameActions?
+	public object whenTurnIsStarted;
```
Comment (AesteroidBlues) [inline Assets/Scripts/Player/Player.cs]: `whenTurnIsStarted` should probably be an event on each card - then when the turn starts for a player we can just loop over all their cards and call that event

File: Assets/Scripts/Player/Player.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		}
+	}
+
+	/**
+	 *  @function StartTurn
+	 *  @description Called at the start of a player's turn.
+	 */
+	public void StartTurn(){
+		// Reset mana counter
+		currentMana = manaThisTurn;
```
Comment (AesteroidBlues) [inline Assets/Scripts/Player/Player.cs:54]: I think we need to increment `manaThisTurn` by one before this, right?

File: Assets/Scripts/Cards/Card.cs [hunk 3]
```diff
@@ -16,6 +16,8 @@ public abstract class Card : ScriptableObject {
 	public int attackPower;
 	public int health;
 
+	public bool isActive;
```
Comment (AesteroidBlues) [inline Assets/Scripts/Cards/Card.cs]: Is this set to true when the card is on the battlefield?

**Review by AesteroidBlues — COMMENTED** [review]
Lookin' good

File: Assets/Scripts/Player/Player.cs (hunk 2 above)
Comment (glitchwolfe) [reply Assets/Scripts/Player/Player.cs:54]: @AesteroidBlues Yes, though I was thinking that would go into a GameBoard class that manages turns/the game session.

File: Assets/Scripts/Cards/Card.cs (hunk 3 above)
Comment (Stralor) [reply Assets/Scripts/Cards/Card.cs]: Moreover, does this conflict with any Unity properties? Let's consider a more precise name.

File: Assets/Scripts/Player/Player.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+		// Draw a card
+		if(hand.Count < maxHandSize){
+			hand.Add(deck.DrawCard());
+		}
```
Comment (Stralor) [inline Assets/Scripts/Player/Player.cs:59]: I haven't looked at the docs. No self-milling in the design?

**Review by Stralor — APPROVED** [review]
Nothing that's gonna create problems yet. Some stubs and implementations (like `whenTurnIsStarted` event as Alex referenced) ought to be fleshed out more.

File: Assets/Scripts/Player/Player.cs (hunk 4 above)
Comment (glitchwolfe) [reply Assets/Scripts/Player/Player.cs:59]: Nope, as of now drawing can either be automatic or (potentially) triggered by card abilities.

File: Assets/Scripts/Player/Player.cs [hunk 5]
```diff
@@ -61,13 +58,10 @@ public class Player : MonoBehaviour {
...
-		// Set all player's inactive cards to active
-		// for( card in cardsOnField ){
-		//    card.isActive = true;
-		// }
-
-		// Trigger any events
-		// whenTurnIsStarted
+		for(card in cardsOnField){
+		   card.isActive = true;   // Set all player's inactive cards to active
+				card.whenTurnIsStarted; // Trigger any events this card has on turn start
```
Comment (AesteroidBlues) [inline Assets/Scripts/Player/Player.cs]: This code doesn't do anything as written, but we don't have a mechanism yet to cause effects based on GameActions so don't worry about it for now

Comment (glitchwolfe) [reply Assets/Scripts/Player/Player.cs]: Ahh true, shoulda commented that one line out ;)

**Pull request merged**